### PR TITLE
[Nightly tests] Increase wait for nodes timeout

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Although there's enough quota, it is possible the AWS doesn't have enough capacity to start up new nodes. According to @allenyin55, the current wait for node timeout is too short. This PR increases the timeout to 3000 seconds (50 minutes) from 600 seconds. Let's see if this can resolve the issue. If it makes things worse, I will revert it quickly (I will closely monitor the infra failure rate)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_tests.yaml [hunk 1]
```diff
@@ -3471,107 +3380,397 @@
...
+    timeout: 3600
+    script: python distributed/test_many_pgs.py
+    wait_for_nodes:
+      num_nodes: 65
+
+    type: job
+    file_manager: job
+  stable: false
+
+- name: k8s_many_pgs_smoke_test
```
Comment (jjyao) [inline release/release_tests.yaml:3629]: ?

Comment (rkooo567) [reply release/release_tests.yaml:3629]: Hmm idk how these are added back lol... Let me fix it

Comment (rkooo567) [reply release/release_tests.yaml]: Fixed

**Comment by rkooo567** [comment]
I will run nightly tests before merging it if approved

**Review by jjyao — APPROVED** [review]

**Comment by rkooo567** [comment]
Verified it worked with nightly tests manually. The failed release-test-pr is because it couldn't find a match.

**Pull request merged**